frontend/static/js/app.js

```diff
@@ -51,39 +53,32 @@ Vue.component('correction-view', {
     },
     fetchJsonData: async function (url, body) {
       this.loading = true;
-      let resp = null;
-      try {
-        resp = await fetch(url, {
-          method: body ? 'POST' : 'GET',
-          headers: {
-            'Content-Type': 'application/json',
-            Authorization: `token ${this.token}`,
-          },
-          body: body ? JSON.stringify(body) : null
-        })
-        if (resp.ok) {
-          return resp.json();
-        }
-      }
-      finally {
+    
+      return await fetch(url, {
+        method: body ? 'POST' : 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `token ${this.token}`,
+        },
+        body: body ? JSON.stringify(body) : null
+      }).then(res =>  { 

```

Comment:
then is not called in case of network error so we'd need to add a finally clause instead but linear code is simpler than callback based code so I don't think we should be changing the pattern here.

Reply:
Reverted back to original with try/catch